assets/js/modules/reader-revenue-manager/utils/validation.js

```diff
@@ -27,7 +27,7 @@
 export function isValidPublicationID( publicationID ) {
 	return (
 		typeof publicationID === 'string' &&
-		/^[A-Za-z0-9_-]+$/.test( publicationID )
+		/^[a-zA-Z0-9_-]+$/.test( publicationID )
```

Comment:
I've updated the JS regex for consistency with the PHP version and the wider codebase.